screens/payment/TopUp.tsx

```diff
@@ -0,0 +1,157 @@
+import { CompositeScreenProps } from "@react-navigation/native";
+import { NativeStackScreenProps } from "@react-navigation/native-stack";
+import { useState } from "react";
+import { Platform, StyleSheet, View } from "react-native";
+
+import PrimaryButton from "@/components/button/PrimaryButton";
+import SecondaryButton from "@/components/button/SecondaryButton";
+import TextInput from "@/components/input/TextInput";
+import HeaderText from "@/components/text/HeaderText";
+import SubHeaderText from "@/components/text/SubHeaderText";
+import BodyContainer from "@/components/ui/BodyContainer";
+import Colors from "@/constants/color";
+import { InputType } from "@/enum/InputType";
+import { AuthenticatedStackParamList, PaymentStackParamList } from "@/types";
+
+type DoubleButtonProps = {
+  leftAmount: number;
+  rightAmount: number;
+  onButtonClicked: (amount: number) => void;
+  activeButton: number | undefined;
+};
+
+export type TopUpProps = CompositeScreenProps<
+  NativeStackScreenProps<PaymentStackParamList, "TopUp">,
+  NativeStackScreenProps<AuthenticatedStackParamList>
+>;
+
+const DoubleButton: React.FC<DoubleButtonProps> = ({
+  leftAmount,
+  rightAmount,
+  onButtonClicked,
+  activeButton,
+}) => {
+  const leftText = leftAmount === 0 ? "Custom" : `฿ ${leftAmount}`;
+  const rightText = rightAmount === 0 ? "Custom" : `฿ ${rightAmount}`;
+  return (
+    <View style={styles.doubleButtonContainer}>
+      <SecondaryButton
+        title={leftText}
+        onPress={() => onButtonClicked(leftAmount)}
+        outerContainerStyle={styles.doubleButton}
+        buttonStyle={{
+          backgroundColor:
+            activeButton === leftAmount ? Colors.red[400] : Colors.gray[700],
+        }}
+      />
+      <SecondaryButton
+        title={rightText}
+        onPress={() => onButtonClicked(rightAmount)}
+        outerContainerStyle={styles.doubleButton}
+        buttonStyle={{
+          backgroundColor:
+            activeButton === rightAmount ? Colors.red[400] : Colors.gray[700],
+        }}
+      />
+    </View>
+  );
+};
+
+const TopUp: React.FC<TopUpProps> = ({ navigation, route }) => {
+  const [amount, setAmount] = useState<number>(0);
+  const [isManualChooseAmount, setIsManualChooseAmount] =
+    useState<boolean>(false);
+  const [activeButton, setActiveButton] = useState<number>();
+
+  const onPressButton = (amount: number) => {
+    setActiveButton(amount);
+    setAmount(amount);
+    setIsManualChooseAmount(amount === 0 ? true : false);
+  };
+
+  return (
+    <BodyContainer>
+      <SubHeaderText text="Current Balance" />
+      <View style={styles.currentBalance}>
+        <HeaderText text={`฿ ${route.params.balance}`} />
+      </View>
+      {isManualChooseAmount && (
+        <TextInput
+          title="Please enter amount (50 baht minimum)"
+          errorText={
+            Number(amount) < 50 ? "Please top up at least 50 baht." : ""
```

Comment:
+amount instead of Number(amount)?

Reply:
Fixed on [7d6023c](https://github.com/ParkingSystemWithLPR/Napark-application/pull/33/commits/7d6023c81bc08f905c97584935b65d69ed7b5de8)